FreeRTOS/FreeRTOS-Kernel
# Added ability to change task notification index for streambuffers

Description
-----------
This PR adds the modification described in https://forums.freertos.org/t/multiple-message-buffers-per-task-interferences/17663:
> The code change necessary to enable your scenario should be relatively simple though - adding a member to the stream buffer structure that holds the task notification index to use with that stream buffer, and a utility API that sets that index

In short, all calls to `xTaskNotify*` APIs are now using the `*Indexed` version and a new field was added to `StreamBuffer_t` to store the notification value used for that buffer.
Now, this value can be get and set using new functions (`uxStreamBufferGetStreamBufferNotificationIndex` and `vStreamBufferSetStreamBufferNotificationIndex`) to avoid changing the overall API.

Test Steps
-----------

* Create any stream buffer, by default this will use 0 (`tskDEFAULT_INDEX_TO_NOTIFY`) as index.
* Change the index using `vStreamBufferSetStreamBufferNotificationIndex`
  * Of course the configuration should allow this with a proper value of `configTASK_NOTIFICATION_ARRAY_ENTRIES`
  * The value must be changed before any task waits for the buffer, otherwise we may never wake them up
* Use the buffer as normal, it should block and resume as usual
* Sending a notification to the task to a different index should not wake up the task.

Checklist:
----------


  * https://github.com/FreeRTOS/FreeRTOS/pull/1150

Related Issue
-----------

https://forums.freertos.org/t/multiple-message-buffers-per-task-interferences/17663

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
